service/core.go

```diff
@@ -63,20 +65,49 @@ type SetContainerer interface {
 	SetContainer(Host)
 }
 
+type metricsCore struct {
+	scope           tally.RootScope
+	runtimeReporter *metrics.RuntimeReporter
+}
+
+func (mc *metricsCore) Metrics() tally.Scope {
+	return mc.scope
+}
+
+func (mc *metricsCore) RuntimeMetrics() *metrics.RuntimeReporter {
+	return mc.runtimeReporter
+}
+
+type tracerCore struct {
+	tracer       opentracing.Tracer
+	tracerCloser io.Closer
+	tracerConfig jaegerconfig.Configuration
+}
+
+func (tc *tracerCore) Tracer() opentracing.Tracer {
+	return tc.tracer
+}
+
+type loggingCore struct {
```

Comment:
I'm not sure we should do this refactor as part of this diff, it would be much easier to iterate on and discuss as part of a separate diff